### ISSUE TITLE
[Bug]: Error Occurs Upon Registering Despite Entering All Information Correctly

### ISSUE THREAD

**Issue by Ryadav0654** [Trisha-tech/OnlineBookSales#340]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Describe the bug

When attempting to register a new account, after entering all the required information on the registration page (e.g., name, email, password.) and clicking the "Register" button, an error message is displayed, preventing successful registration.

1. Navigate to the registration page.
2. Fill in all mandatory fields (e.g., username, email, password, etc.).
3. Click the "Register" button.
4. Observe that an error message appears and the registration fails.

### Expected behavior

Upon entering all required information correctly and clicking the "Register" button, the user should be successfully registered and redirected to the appropriate page (e.g., login page or dashboard).

### Add ScreenShots

![Screenshot 2024-10-07 013908](https://github.com/user-attachments/assets/cbb174d1-1f16-47b2-bb36-cb6f63d7003d)

### On which device are you experiencing this bug?

Windows

### Record

- [X] I have read the Contributing Guidelines
- [X] I'm a GSSOC'24 contributor
- [X] I have starred the repository

**Comment by Ryadav0654** [comment]
Hi @Trisha-tech ,

I've reported an issue related to the registration process, where users encounter an error after entering all the required information. Could you please review the issue and assign it to me for resolution?

Thanks!

**Comment by Suman373** [comment]
> Hi @Trisha-tech ,
> 
> I've reported an issue related to the registration process, where users encounter an error after entering all the required information. Could you please review the issue and assign it to me for resolution?
> 
> Thanks!

Did you run your server on localhost while testing the client side ?

**Comment by Ryadav0654** [comment]
> > Hi @Trisha-tech ,
> > I've reported an issue related to the registration process, where users encounter an error after entering all the required information. Could you please review the issue and assign it to me for resolution?
> > Thanks!
> 
> Did you run your server on localhost while testing the client side ?

No, I didn't run the server on localhost.

**Comment by Suman373** [comment]
> > > Hi @Trisha-tech ,
> > > I've reported an issue related to the registration process, where users encounter an error after entering all the required information. Could you please review the issue and assign it to me for resolution?
> > > Thanks!
> > 
> > 
> > Did you run your server on localhost while testing the client side ?
> 
> No, I didn't run the server on localhost.

If you dont the run the server, how will your api request be successful

**Comment by Suman373** [comment]
> > If you dont the run the server, how will your api request be successful
> 
> The issue is on your live website. When I visited the live site at https://book4u-j5au.onrender.com/signup, it gave an error.

Obviously. The api endpoint is hardcoded with localhost in SignUp page. There is no env provided for the live website. Do you want to fix this?
